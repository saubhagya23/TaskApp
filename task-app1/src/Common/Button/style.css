.container {
  box-sizing: border-box;
  padding: 10px;
  margin: 10px;
  background-color: darkgray;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 5px;
}