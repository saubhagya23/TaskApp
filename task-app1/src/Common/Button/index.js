import React, { Component } from 'react';
import './style.css';

class Button extends Component {
  render() {
    const {
      btnText,
      btnStyle,
      onBtnClick,
    } = this.props;
    
    return(
      <div className={`container ${btnStyle}`} onClick={onBtnClick} data-testid="button">
        {btnText} 
      </div>
    );
  }
}

export default Button;