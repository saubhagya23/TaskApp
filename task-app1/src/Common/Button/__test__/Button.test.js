import React from 'react';
import ReactDOM from 'react-dom';
import Button from './../index';
// import { render } from '../../../utils/test-utils'
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders button without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button/>, div)
})

it("renders button correctly", () => {
  const { getByTestId } = render(<Button btnText="click me" />);
  expect(getByTestId('button')).toHaveTextContent("click me")
})

it("renders button correctly", () => {
  const { getByTestId } = render(<Button btnText="logout" />);
  expect(getByTestId('button')).toHaveTextContent("logout")
})

it("matches snapshot", () => {
  const buttonTree = renderer.create(<Button btnText="click me"></Button>).toJSON();
  expect(buttonTree).toMatchSnapshot();
})