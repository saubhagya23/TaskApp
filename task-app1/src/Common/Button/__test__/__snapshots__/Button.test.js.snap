exports[`test matches snapshot 1`] = `
<div
  className="container undefined"
  data-testid="button"
  onClick={undefined}>
  click me
</div>
`;
